src/daft-core/src/array/ops/list.rs

```diff
@@ -35,6 +36,31 @@ fn join_arrow_list_of_utf8s(
         })
 }
 
+fn arrow_list_contains(
+    list_elements: impl Iterator<Item = Option<Box<dyn arrow2::array::Array>>>,
+    elements: impl Iterator<Item = Box<dyn arrow2::array::Array>>,
+) -> DaftResult<arrow2::array::BooleanArray> {
+    let mut contains_elements: Vec<Option<bool>> = Vec::new();
+    for (arr, element) in list_elements.zip(elements) {
```

Comment:
can run this as a map and give an iterator then then create BooleanArray with `from_iter`: https://github.com/jorgecarleitao/arrow2/blob/main/src/array/boolean/from.rs#L12

Reply:
Oh yes, I was going to do that and forgot... Thanks!

Reply:
Hmm actually, I'm unable to do this because the code that executes in the Iterator can potentially return a DaftError (we run a `.equals` in that code).

The iterator that returns thus can't be piped into BooleanArray. The only way I could find to make it work still requires intermediate memory:

```python
let results: Result<Vec<_>> = list_elements.zip(elements).map(...).collect();
Ok(BooleanArray::from(results?))
```

Is there a better way?